src/modules/dashboard/sagas/actions/smite.ts

```diff
@@ -0,0 +1,193 @@
+
+    const {
+      emitDomainReputationPenalty,
+      annotateEmitDomainReputationPenalty,
+    } = yield createTransactionChannels(metaId, [
+      'emitDomainReputationPenalty',
+      'annotateEmitDomainReputationPenalty',
+    ]);
+
+    const createGroupTransaction = ({ id, index }, config) =>
+      fork(createTransaction, id, {
+        ...config,
+        group: {
+          key: batchKey,
+          id: metaId,
+          index,
+        },
+      });
+
+    yield createGroupTransaction(emitDomainReputationPenalty, {
+      context: ClientType.ColonyClient,
+      methodName: 'emitDomainReputationPenaltyWithProofs',
+      identifier: colonyAddress,
+      params: [domainId, userAddress, amount],
+      ready: false,
+    });
+
+    if (annotationMessage) {
+      yield createGroupTransaction(annotateEmitDomainReputationPenalty, {
+        context: ClientType.ColonyClient,
+        methodName: 'annotateTransaction',
+        identifier: colonyAddress,
+        params: [],
+        ready: false,
+      });
+    }
+
+    yield takeFrom(
+      emitDomainReputationPenalty.channel,
+      ActionTypes.TRANSACTION_CREATED,
+    );
+    if (annotationMessage) {
+      yield takeFrom(
+        annotateEmitDomainReputationPenalty.channel,
+        ActionTypes.TRANSACTION_CREATED,
+      );
+    }
+
+    yield put(transactionReady(emitDomainReputationPenalty.id));
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      emitDomainReputationPenalty.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+
+    yield takeFrom(
+      emitDomainReputationPenalty.channel,
+      ActionTypes.TRANSACTION_SUCCEEDED,
+    );
+
+    if (annotationMessage) {
+      yield put(transactionPending(annotateEmitDomainReputationPenalty.id));
+
+      let annotationMessageIpfsHash = null;
+      annotationMessageIpfsHash = yield call(
+        ipfsUpload,
+        JSON.stringify({
+          annotationMessage,
+        }),
+      );
+
+      yield put(
+        transactionAddParams(annotateEmitDomainReputationPenalty.id, [
+          txHash,
+          annotationMessageIpfsHash,
+        ]),
+      );
+
+      yield put(transactionReady(annotateEmitDomainReputationPenalty.id));
+
+      yield takeFrom(
+        annotateEmitDomainReputationPenalty.channel,
+        ActionTypes.TRANSACTION_SUCCEEDED,
+      );
+    }
+
+    yield apolloClient.query<UserReputationQuery, UserReputationQueryVariables>(
+      {
+        query: UserReputationDocument,
+        variables: {
+          colonyAddress,
+          address: userAddress,
+          domainId,
+        },
+        fetchPolicy: 'network-only',
+      },
+    );
+
+    yield apolloClient.query<UserReputationQuery, UserReputationQueryVariables>(
+      {
+        query: UserReputationDocument,
+        variables: {
+          colonyAddress,
+          address: AddressZero,
+          domainId,
+        },
+        fetchPolicy: 'network-only',
+      },
+    );
```

Comment:
Why are we calling this refresh query with `AddressZero` ?

Reply:
Like Olga already mentioned, the UserReputation is called with `AddressZero` when we want to fetch the colony's total reputation, which should change after the penalization.